### ISSUE TITLE
Recurring notifications don't advance note time

### ISSUE THREAD

**Issue by keithcrone** [orgzly/orgzly-android#84]
If I schedule a recurrence, say every hour, the notification works every hour (woohoo) but if I just dismiss the note itself is not updated to the new notification (or scheduled) time, as would occur if you manually acknowledged the event from the note itself.

**Comment by MackieLoeffel** [comment]
I think, this is the correct behavior. Dismissing a note should not mark a note as done in my opinion. What is missing is a done-button in the notification, which would advance the notification time, as mentioned here: https://github.com/orgzly/orgzly-android/issues/26#issuecomment-285449871

**Comment by keithcrone** [comment]
But why receive notifications for an event I haven't acknowledged I want to occur?

I set a note to notify me today at 1012 that has notified me every hour on the dot, yet the note is still scheduled for 1012. Prior to this I manually acknowledged the event to advance the time to the next schedule in the interval. Now there are notifications regardless of that acknowledgement.

**Comment by nevenz** [comment]
Are you an Org mode user? I can see this being confusing for the one who is not. Although I'm not sure that the current behaviour is correct anyway.

> But why receive notifications for an event I haven't acknowledged I want to occur?

Didn't you do that by setting the repeater in the first place?

For reminders, using a repeater is equivalent to setting multiple times.

```
<2017-05-17 Tue +3d>
```

is the same as:

```
<2017-05-17 Tue>
<2017-05-20 Tue>
<2017-05-23 Tue>
...
```

( This is not when *scheduled* times are shown in Org mode agenda - they'd be shown *always* after initial time, as long as the note is not done. Current behaviour for reminders in Orgzly is more similar to plain timestamps in Org mode. Assuming we even want reminders to be agenda-in-your-face. )

What you want is probably setting the time *without* repeater, but then being able to re-set it when reminder is shown. But that's not what the repeater is for. It could be, but that would break expected behaviour for most Org mode users. It could be a new option though. Type of repeater (`.+`, `++`, `+`) might also be important.

Bottom line - I think we have yet to define all this for Orgzly - when to show reminders, shift times etc.

**Comment by keithcrone** [comment]
Then I don't get it. I would expect notification to occur at the next timestamp. I wouldn't get the next timestamp unless I marked this interval as DONE. You're saying I'll get notification at the next regular interval regardless.

**Comment by nevenz** [comment]
> You're saying I'll get notification at the next regular interval regardless.

Yes, that's how it currently works. Time is just the *starting* time from which you'll start getting reminders every interval.

It's for things you want to do periodically, regardless of whether you've done them the last time or not. For example `<2017-01-01 .+2d> Go for a run`. Starting time doesn't matter, you want to do that every other day.

But what you expect is also a valid use case of course and needs to be supported.

**Comment by keithcrone** [comment]
What would I use that for? Without know the next (or even last) notification time, why would that be useful?

**Comment by nevenz** [comment]
> What would I use that for? Without know the next (or even last) notification time, why would that be useful?

In the example I gave (`<2017-01-01 .+2d> Go for a run`), it would be useful to know how lazy I am.

I guess the way it currently works is kinda Org mode-centric, where you do explicitly mark the note as done to shift the time. It's not shifted after it's being displayed in the agenda for the first time.

A new preference can be added to change the current behaviour - if the time has a repeater and reminder is displayed, shift time. But there are different types of a repeaters. And you probably don't want `+` for reminders here, but `.+` or `++`. So perhaps only shift for those two if reminder is shown.

Sounds messy. I'll see to create a Wiki page or something and list all the possible scenarios. I'm also curious how others do it (like `org-alert`).

**Comment by keithcrone** [comment]
When I started the issue, I thought your plan was to advance the TODO interval since the notifications kept coming, but that doesn't seem to be the case.

All I think anyone would expect is to be alerted on a <Scheduled> timestamp. I really wouldn't expect Orgzly to do anything else but alert me when I have something scheduled.

If I want to advance the TODO to the next interval, I have to do so manually. Same thing in Emacs.

If you think an alert isn't always appropriate, make alerts specific to the Note, maybe with a specific Property pair value like [Alert:True or False]. Specific properties wouldn't break anything in Emacs correct?

If you're concerned about your level of lazy, make the notification persistant so it can't be swiped away. 

[Thank you again for putting out a good product. I apologize for being a pain about these. I spend  most of time on mobile so Notifications are a HUGE thing for me]

**Comment by nevenz** [comment]
> All I think anyone would expect is to be alerted on a timestamp. I really wouldn't expect Orgzly to do anything else but alert me when I have something scheduled.

OK, this makes sense. I'll change the behaviour so that notification is only displayed for the exact time, basically ignoring repeaters, so we'll see how that goes.

I'll also add a button (checkmark) for marking the note as done, so user can shift the time (or actually mark the note as done, if there is no repeater).

**Comment by nevenz** [comment]
Hmm, but how to actually represent the repeating reminder then, which should be a feature too?

I'd like to avoid Orgzly-specific properties as long as possible, but I don't see Org timestamp being enough here.

**Comment by keithcrone** [comment]
If I create a repeating event in Google Calendar, I can see all future events, with or without an alert.
If I create a repeating Reminder (in Calendar, Keep, or Inbox), I can also see all future events, regardless of whether I acknowledge the current interval.

Let me ask this: if I create a simple repeater daily at 8am, what should I see when I search something like s.10d? Currently I would see just the single event.

It sounds like you want something closer to org-clone-subtree-with-time-shift. Maybe offer an option for the number of tines to repeat the event [using the same picker for the repeater interval]. Then the same s.10d search would show the next 10 reminders.

**Comment by nevenz** [comment]
> If I create a repeating event in Google Calendar, I can see all future events, with or without an alert.
If I create a repeating Reminder (in Calendar, Keep, or Inbox), I can also see all future events, regardless of whether I acknowledge the current interval.

> Let me ask this: if I create a simple repeater daily at 8am, what should I see when I search something like s.10d? Currently I would see just the single event.

Search just searches for *notes* and `s` just looks at the date/time part, so that's correct.

More relevant is *agenda support* (#55) - a calendar-like display where you would (optionally) see the note for every day.

> It sounds like you want something closer to org-clone-subtree-with-time-shift. Maybe offer an option for the number of tines to repeat the event [using the same picker for the repeater interval]. Then the same s.10d search would show the next 10 reminders.

It's an idea for a new feature for those who need it, but I wouldn't want to list all the possible times as a way to specify repeating time. Nor we could, if there's no end time.

**Comment by nevenz** [comment]
Behaviour has been changed in 7bdece1, let's move any further discussion to #26.
